johnsonandjohnson/Bodiless-JS
# POC : Vital Decoupled with Card/Buttons/Links [canary]

## Changes
- Separate out the packages of vital-card, vital-button, vital-link from design & editable.
  - vital-card-core, vital-button-core, vital-link-core now contain the vital design only.
     -  added "asExternalBaseCOMPONENT" token stub that sets data uses withDesign to populate data via addProps.  Could be used as is or eventually replaced/set by vital-component-CMS package. 
  - Existing vital-card, vital-buttons, vital-link add the editing schemas and hydration features and function identical to before (thus no breaking changes)
    - For each package -- export the core collection followed by any tokens that need the editing. (Which includes anything without prefixed "with")  and add editable/flowcontainer tokens as needed.)
- Add sites/gatsby-vanilla site that only includes @bodiless/fclasses, @bodiless/vital-*-core & @bodiless/vital-elements and none of the 'bodiless editable funcitonality'.   
  - Builds with Bodiless tailwind scripts.  
  - Demos vital design with no editing functionality

## Test Instructions
- No Change in Vital Bodiless Demo and functionality
   - npm run start on sites/vital-demo and visit http://localhost:8000/styleguide/card -- confirm no change. 
   - verify npm run build & serve on sites/vital-demo and visit http://localhost:9000/styleguide/card -- confirm no change. 
- Gatsby Vanilla Site demo'ing vital only design + tailwind building from bodiless.
   - npm run start on sites/gatsby-vanilla and visit http://localhost:8000/styleguide/card -- there vital design cards should be displayed with stubbed external data. 
   - verify npm run build & serve on sites/gatsby-vanilla and visit http://localhost:9000/styleguide/card
